src/finiteVolume/processes/LES.cpp

```diff
@@ -0,0 +1,251 @@
+#include "LES.hpp"
+#include "finiteVolume/compressibleFlowFields.hpp"
+#include "utilities/mathUtilities.hpp"
+#include "utilities/petscError.hpp"
+
+ablate::finiteVolume::processes::LES::LES(std::shared_ptr<eos::transport::TransportModel> transportModelIn) : transportModel(std::move(transportModelIn)) {}
+
+//?? I have registered separate sources for momentum and energy fluxes. any problem??//
+void ablate::finiteVolume::processes::LES::Setup(ablate::finiteVolume::FiniteVolumeSolver& flow) {
+    if (transportModel) {
+        if (diffusionData.diffFunction.function) {
+            // Register the euler/Momentum LESdiffusion source term
+            flow.RegisterRHSFunction(LESMomentumFlux,
+                                     &diffusionData,
+                                     CompressibleFlowFields::EULER_FIELD,
+                                     {CompressibleFlowFields::EULER_FIELD, CompressibleFlowFields::DENSITY_EV_FIELD},
+                                     {CompressibleFlowFields::VELOCITY_FIELD});
+            // Register the euler/Energy LESdiffusion source term
+            flow.RegisterRHSFunction(LESEnergyFlux,
+                                     &diffusionData,
+                                     CompressibleFlowFields::EULER_FIELD,
+                                     {CompressibleFlowFields::EULER_FIELD, CompressibleFlowFields::DENSITY_EV_FIELD},
+                                     {CompressibleFlowFields::TEMPERATURE_FIELD, CompressibleFlowFields::VELOCITY_FIELD});
+            // Register the Species LESdiffusion source term
+            flow.RegisterRHSFunction(LESSpeciesFlux,
+                                     &diffusionData,
+                                     CompressibleFlowFields::DENSITY_YI_FIELD,
+                                     {CompressibleFlowFields::DENSITY_YI_FIELD, CompressibleFlowFields::DENSITY_EV_FIELD},
+                                     {CompressibleFlowFields::EV_FIELD});
+            // Register the ev LESdiffusion source term
+            flow.RegisterRHSFunction(
+                LESevFlux, &diffusionData, CompressibleFlowFields::EV_FIELD, {CompressibleFlowFields::EULER_FIELD, CompressibleFlowFields::DENSITY_EV_FIELD}, {CompressibleFlowFields::EV_FIELD});
+        }
+    }
+
+    diffusionData.computeTemperatureFunction = eos->GetThermodynamicFunction(eos::ThermodynamicProperty::Temperature, flow.GetSubDomain().GetFields());
+}
+/*?? I use velocity and temperature in two of my functions, LESEnergyFlux and CompressibleFlowComputeLESStressTensor, should I compute those fields using "UpdateAux.." function or those fields get
```

Comment:
They are updated in ns.  If you plan on having that process active it should be ok.